src/index.ts

```diff
@@ -1,8 +1,8 @@
 import { Observable, Subscription, BehaviorSubject } from 'rxjs'
 
-export interface Connection {
+export interface Connection<T extends string | ArrayBuffer | Blob = string | ArrayBuffer | Blob> {
```

Comment:
`ArrayBuffer` appears twice and also not sure why both `Blob` and `Blob = string` are specified. Not sure what the latter means.

Reply:
Operators precedence with a bit different here, `=` takes higher place. Essentially it's generic argument `T` with a constraint of `string | ArrayBuffer | Blob` **and** default value of `string | ArrayBuffer | Blob`. Without the latter you'll have to specify generic argument every time.

Reply:
Doah of course, don't suppose it's possible to add `()` around the union types to make that more clear?

Reply:
And yeah I get that `=` has lower precedence (rather than higher) now, I should have really realised that in the first place ;)

Reply:
> don't suppose it's possible to add () around the union types to make that more clear?

I didn't know that, but it is in fact possible! Added parenthesis.

> And yeah I get that = has lower precedence

Yeah, you're right, sorry for messing it up even more :)